engine/sync.go

```diff
@@ -262,6 +254,18 @@ func newSyncServer(e *Engine) *syncServer {
 	}
 }
 
+// newSession allocates a new session ID and starts managing the session with
+// the provided node.
+func (s *syncServer) newSession(node net.IP) *syncSession {
+	s.sessionLock.Lock()
+	session := newSyncSession(node, s.nextSessionID)
+	s.nextSessionID++
+	s.sessions[session.id] = session
+	s.sessionLock.Unlock()
```

Comment:
Why aren't the `Unlock()` calls deferred?

Reply:
SGTM, done.